src/Microsoft.Azure.SignalR.Common/ClientInvocation/ClientResultsManager.cs

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET7_0_OR_GREATER
+using System.Threading;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using System.Buffers;
+using System;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ClientResultsManager : IClientResultsManager, IInvocationBinder
+    {
+        private readonly ConcurrentDictionary<string, (Type Type, string ConnectionId, object Tcs, Action<object, CompletionMessage> Complete)> _pendingInvocations = new();
+        private ulong _lastInvocationId = new();
+
+        private readonly ConcurrentDictionary<string, (Type Type, string ConnectionId, string CallerServerId, object Tcs, Action<object, CompletionMessage> Complete)> _routedInvocations = new();
```

Comment:
so long parameters, extract to a record?

Reply:
fixed